webapp/src/extensionsBrowser.tsx

```diff
@@ -0,0 +1,453 @@
+                    <div className={"extensionTag " + (selectedTag == c ? "selected" : "")} onClick={() => handleCategoryClick(c)}>{c}</div>
+                )}
+            </div>
+            {/* <div className="importButton">
+                <span>{lf("or ")}</span>
+                <div className="importButtonLink" onClick={() => setShowImportExtensionDialog(true)}>{lf("import extension")}</div>
+            </div> */}
+        </div>
+        <div className="extension-display">
+            {selectedTag != "" || searchFor != "" ?
+                <div className="breadcrumbs">
+                    <span className="link" onClick={handleHomeButtonClick}>{lf("Home")}</span>
+                    {selectedTag != "" ? <span>/</span> : undefined}
+                    {selectedTag != "" ? <span>{selectedTag}</span> : undefined}
+                </div> :
+                searchFor == "" ?
+                    <div className="tab-header">
+                        <Button title={lf("Recommended")} label={lf("Recommended")} onClick={() => { setCurrentTab(TabState.Recommended) }} className={currentTab == TabState.Recommended ? "selected" : ""} />
+                        <Button title={lf("Installed")} label={lf("Installed")} onClick={() => { setCurrentTab(TabState.Installed) }} className={currentTab == TabState.Installed ? "selected" : ""} />
+                        <Button title={lf("In Development")} label={lf("In Development")} onClick={() => { setCurrentTab(TabState.InDevelopment) }} className={currentTab == TabState.InDevelopment ? "selected" : ""} />
+                    </div> :
+                    undefined
+            }
+            {selectedTag != "" || searchFor != "" ?
+                <div className="extension-grid">
+                    {extensionsToShow?.map(scr =>
+                        <ExtensionCard scr={scr} onCardClick={installExtension} learnMoreUrl={scr.fullName ? `/pkg/${scr.fullName}` : undefined}
+                            name={scr.name} imageUrl={scr.imageUrl} description={scr.description} loading={scr.loading} />)}
+                </div> :
+                currentTab == TabState.Recommended ? preferredExts.map(e => <ExtensionCard scr={e} name={e.name} onCardClick={installExtension} imageUrl={e.imageUrl} description={e.description}
+                        learnMoreUrl={e.fullName ? `/pkg/${e.fullName}`: undefined} loading={e.loading} />) :
+                    currentTab == TabState.Installed ? installedExtensions.map(e =>
+                        <ExtensionCard scr={e} name={e.name} onCardClick={() => handleInstalledCardClick(e)} imageUrl={e.imageUrl} description={e.description} learnMoreUrl={e.fullName ? `/pkg/${e.fullName}` : undefined}/>
+                    ) :
+                        local.forEach(p => {
+                            <ExtensionCard scr={p} name={p.name} description={lf("Local copy of {0} hosted on github.com", p.githubId)} onCardClick={addLocal} />
+                        })
+            }
```

Comment:
these nested ternaries are a little hard to parse. is there any way to break them up?